libapicore/ApiServer.cpp

```diff
@@ -0,0 +1,69 @@
+#include "ApiServer.h"
+#include "BuildInfo.h"
+
+ApiServer::ApiServer(AbstractServerConnector *conn, serverVersion_t type, Farm &farm, bool &readonly) : AbstractServer(*conn, type), m_farm(farm)
+{
+	this->bindAndAddMethod(Procedure("miner_getstat1", PARAMS_BY_NAME, JSON_OBJECT, NULL), &ApiServer::getMinerStat1);
+	if (!readonly) {
+		this->bindAndAddMethod(Procedure("miner_restart", PARAMS_BY_NAME, JSON_OBJECT, NULL), &ApiServer::doMinerRestart);
+		this->bindAndAddMethod(Procedure("miner_reboot", PARAMS_BY_NAME, JSON_OBJECT, NULL), &ApiServer::doMinerReboot);
+	}
+}
+
+void ApiServer::getMinerStat1(const Json::Value& request, Json::Value& response)
+{
+	(void) request; // unused
+	
+	auto runningTime = std::chrono::duration_cast<std::chrono::minutes>(steady_clock::now() - this->m_started);
+	
+	SolutionStats s = this->m_farm.getSolutionStats();
+	WorkingProgress p = this->m_farm.miningProgress();
+	
+	ostringstream totalMhEth; 
+	ostringstream totalMhDcr; 
+	ostringstream detailedMhEth;
+	ostringstream detailedMhDcr;
+	ostringstream tempAndFans;
+	ostringstream invalidStats;
+	
+	totalMhEth << std::fixed << std::setprecision(0) << (p.rate() / 1000.0f) << ";" << s.getAccepts() << ";" << s.getRejects();
+	totalMhDcr << "0;0;0"; // DualMining not supported
+	invalidStats << s.getFailures() << ";0"; // Invalid + Pool switches
+	invalidStats << ";0;0"; // DualMining not supported
+	
+	int gpuIndex = 0;
+	int numGpus = p.minersHashes.size();
+	for (auto const& i: p.minersHashes)
+	{
+		detailedMhEth << std::fixed << std::setprecision(0) << (p.minerRate(i) / 1000.0f) << (((numGpus -1) > gpuIndex) ? ";" : "");
+		detailedMhDcr << "off" << (((numGpus -1) > gpuIndex) ? ";" : ""); // DualMining not supported
+		tempAndFans << "50;50" <<(((numGpus -1) > gpuIndex) ? ";" : ""); // Fetching Temp and Fans not supported
+		gpuIndex++;
+	}
+
+	response[0] = ETH_PROJECT_VERSION;           //miner version.
+	response[1] = toString(runningTime.count()); // running time, in minutes.
+	response[2] = totalMhEth.str();              // total ETH hashrate in MH/s, number of ETH shares, number of ETH rejected shares.
```

Comment:
This is in KH/s for me. Is this a typo?

-- 0.15.0.dev0

Reply:
comments are taken 1:1 of claymores documentation. Values itself are also the same as if it would be from claymore.

For reference, the api is modeled after claymores api to use existing monitoring software

Reply:
But it shows thousands of megahashes that I don't have. And is 1000 * the number reported by ethminer on console. If claimore's comment is wrong why should we keep it wrong here as well?

Reply:
> comments are taken 1:1 of claymores documentation. Values itself are also the same as if it would be from claymore.
> 
> For reference, the api is modeled after claymores api to use existing monitoring software

Hey @smurfy Thank's for the continuous stream of updates! I'm trying to use this Monitoring setup (https://github.com/bobdobs/claymore-ethereum-log-genereator) I already put in many hours in trying to figure out why it won't work. I'm just learning about JSON-RPC 2.0, Socks, curl, netcat and so on. Could yo may be point me in the right direction what I have to change ? Thank's